cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -372,31 +372,32 @@ func ValidateHostPort(endpoint string, fldPath *field.Path) field.ErrorList {
 // ValidateIPNetFromString validates network portion of ip address
 func ValidateIPNetFromString(subnetStr string, minAddrs int64, isDualStack bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if isDualStack {
-		subnets, err := utilnet.ParseCIDRs(strings.Split(subnetStr, ","))
+	subnets, err := utilnet.ParseCIDRs(strings.Split(subnetStr, ","))
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, subnetStr, "couldn't parse subnet"))
+		return allErrs
+	}
+	switch {
+	// if DualStack and more than one cidr validate if there is at least one of each IP family
+	case isDualStack && len(subnets) > 1:
```

Comment:
we should decide if we allow more than 2 and we use only the first from each family, `len(subnets) > 1`
or we just only allow 2, 1 from each family ' len(subnets) ==2`

Reply:
allowing more than two and arbitrarily skipping/ignoring some is really confusing… that doesn't seem like a good idea

Reply:
Let's allow for two (but no more).